com.vrcfury.vrcfury/Editor/VF/Service/OnDemandSyncService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using VF.Builder;
+using VF.Feature.Base;
+using VF.Injector;
+using VF.Utils.Controller;
+using VRC.SDK3.Avatars.Components;
+
+namespace VF.Service {
+    [VFService]
+    public class OnDemandSyncService {
+        [VFAutowired] private readonly AvatarManager manager;
+        [VFAutowired] private readonly MathService math;
+        [VFAutowired] private readonly DirectBlendTreeService directTree;
+        [VFAutowired] private readonly SmoothingService smoothingService;
+        
+        private readonly List<VFAFloat> onDemandSyncParameters = new List<VFAFloat>();
+        
+        public void SetParameterOnDemandSync(VFAFloat parameter) {
+            onDemandSyncParameters.Add(parameter);
+        }
+
+        [FeatureBuilderAction(FeatureOrder.OnDemandSync)]
+        public void Apply() {
+            if (onDemandSyncParameters.Count == 0) return;
+            
+            var fx = manager.GetController(VRCAvatarDescriptor.AnimLayerType.FX);
+            var layers = fx.GetRaw().GetLayers();
+            foreach (var param in onDemandSyncParameters) {
+                manager.GetParams().GetParam(param.Name()).networkSynced = false;
```

Comment:
This isn't old-vrcsdk compatible. Check ControllerManager.cs for an example of how this is accessed using reflection. On old vrcsdks, a warning should be printed and the entire on demand sync skipped.